ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/BeaconStateImpl.java

```diff
@@ -0,0 +1,719 @@
+import tech.pegasys.artemis.util.backing.ViewRead;
+import tech.pegasys.artemis.util.backing.tree.TreeNode;
+import tech.pegasys.artemis.util.backing.type.BasicViewTypes;
+import tech.pegasys.artemis.util.backing.type.ContainerViewType;
+import tech.pegasys.artemis.util.backing.type.ListViewType;
+import tech.pegasys.artemis.util.backing.type.VectorViewType;
+import tech.pegasys.artemis.util.backing.view.AbstractBasicView;
+import tech.pegasys.artemis.util.backing.view.BasicViews.BitView;
+import tech.pegasys.artemis.util.backing.view.BasicViews.Bytes32View;
+import tech.pegasys.artemis.util.backing.view.BasicViews.UInt64View;
+import tech.pegasys.artemis.util.backing.view.ContainerViewImpl;
+import tech.pegasys.artemis.util.backing.view.ViewUtils;
+import tech.pegasys.artemis.util.config.Constants;
+
+public class BeaconStateImpl extends ContainerViewImpl<BeaconStateImpl>
+    implements MutableBeaconState, BeaconStateCache {
+
+  // The number of SimpleSerialize basic types in this SSZ Container/POJO.
+  public static final int SSZ_FIELD_COUNT = 14;
+
+  public static final ContainerViewType<BeaconStateImpl> TYPE =
+      new ContainerViewType<>(
+          List.of(
+              BasicViewTypes.UINT64_TYPE,
+              BasicViewTypes.UINT64_TYPE,
+              Fork.TYPE,
+              BeaconBlockHeader.TYPE,
+              new VectorViewType<>(
+                  BasicViewTypes.BYTES32_TYPE, Constants.SLOTS_PER_HISTORICAL_ROOT),
+              new VectorViewType<>(
+                  BasicViewTypes.BYTES32_TYPE, Constants.SLOTS_PER_HISTORICAL_ROOT),
+              new ListViewType<>(BasicViewTypes.BYTES32_TYPE, Constants.HISTORICAL_ROOTS_LIMIT),
+              Eth1Data.TYPE,
+              new ListViewType<>(Eth1Data.TYPE, Constants.SLOTS_PER_ETH1_VOTING_PERIOD),
+              BasicViewTypes.UINT64_TYPE,
+              new ListViewType<>(ValidatorImpl.TYPE, Constants.VALIDATOR_REGISTRY_LIMIT),
+              new ListViewType<>(BasicViewTypes.UINT64_TYPE, Constants.VALIDATOR_REGISTRY_LIMIT),
+              new VectorViewType<>(
+                  BasicViewTypes.BYTES32_TYPE, Constants.EPOCHS_PER_HISTORICAL_VECTOR),
+              new VectorViewType<>(
+                  BasicViewTypes.UINT64_TYPE, Constants.EPOCHS_PER_SLASHINGS_VECTOR),
+              new ListViewType<>(
+                  PendingAttestation.TYPE, Constants.MAX_ATTESTATIONS * Constants.SLOTS_PER_EPOCH),
+              new ListViewType<>(
+                  PendingAttestation.TYPE, Constants.MAX_ATTESTATIONS * Constants.SLOTS_PER_EPOCH),
+              new VectorViewType<>(BasicViewTypes.BIT_TYPE, Constants.JUSTIFICATION_BITS_LENGTH),
+              Checkpoint.TYPE,
+              Checkpoint.TYPE,
+              Checkpoint.TYPE),
+          BeaconStateImpl::new);
+
+  @Label("sos-ignore")
+  private final TransitionCaches transitionCaches;
+
+  @Label("sos-ignore")
+  private final boolean builder;
+
+  // Versioning
+  @SuppressWarnings("unused")
```

Comment:
looks like all these unused fields can go away?

Reply:
Absolutely, but later, since SSZ SoS relies on fields via reflection for now